src/Processors/Formats/Impl/TemplateRowInputFormat.cpp

```diff
@@ -566,13 +566,32 @@ void registerTemplateSchemaReader(FormatFactory & factory)
 {
     for (bool ignore_spaces : {false, true})
     {
-        factory.registerSchemaReader(ignore_spaces ? "TemplateIgnoreSpaces" : "Template", [ignore_spaces](ReadBuffer & buf, const FormatSettings & settings)
+        String format_name = ignore_spaces ? "TemplateIgnoreSpaces" : "Template";
+        factory.registerSchemaReader(format_name, [ignore_spaces](ReadBuffer & buf, const FormatSettings & settings)
         {
             size_t index = 0;
             auto idx_getter = [&](const String &) -> std::optional<size_t> { return index++; };
             auto row_format = fillRowFormat(settings, idx_getter, false);
             return std::make_shared<TemplateSchemaReader>(buf, ignore_spaces, fillResultSetFormat(settings), row_format, settings.template_settings.row_between_delimiter, settings);
         });
+        factory.registerAdditionalInfoForSchemaCacheGetter(format_name, [](const FormatSettings & settings)
+        {
+            size_t index = 0;
+            auto idx_getter = [&](const String &) -> std::optional<size_t> { return index++; };
+            auto row_format = fillRowFormat(settings, idx_getter, false);
+            std::unordered_set<FormatSettings::EscapingRule> visited_escaping_rules;
+            String result = fmt::format("row_format={}, resultset_format={}, row_between_delimiter={}",
+                settings.template_settings.row_format,
+                settings.template_settings.resultset_format,
+                settings.template_settings.row_between_delimiter);
+            for (auto escaping_rule : row_format.escaping_rules)
```

Comment:
`const auto &`?

Reply:
`escaping_rule ` is a Enum value, using const reference doesn't make sense, copying it as cheap as creating a reference